Timeline

MMTsallagova opened the pull request:
- add actorDeltaTimeMultiplier
- add game DeltaTimeMultiplier

MMTsallagova committed:
Timeline

- add actorDeltaTimeMultiplier
- add game DeltaTimeMultiplier

nikchernyakov commented on Engine/InGame/Game.cpp:
```diff
@@ -32,3 +32,14 @@ void Game::Render()
     }
 }
+
+void Game::SetGameDeltaTime(float deltaMultiplier)
```
Переименуй deltaMultiplier -> deltaTimeMultiplier
А метод в SetGameDeltaTimeMultiplier, ты же его меняешь

nikchernyakov commented on Engine/InGame/Game.cpp:
```diff
@@ -32,3 +32,14 @@ void Game::Render()
+    gameDeltaTimeMultiplier = deltaMultiplier;
+}
+
+float Game::GetWorldDeltaTime()
```
Моя вина, забыл в трелло переименовать. Нужно GetGameDeltaTime сделать

Jass2100 commented on Engine/InGame/GameObject/Actor.h:
```diff
@@ -18,14 +18,18 @@ class Actor : public GameObject
     Transform* GetTransform() const;
 
     void AddComponent(ActorComponent* component);
+    void setActorDeltaTime(float deltaMultiplier);
```
Public функция с большой буквы

nikchernyakov replied:
У тебя публичный метод с маленькой буквы и называться должен с Multiplier на конце

nikchernyakov commented on Engine/InGame/Game.h:
```diff
@@ -19,11 +19,18 @@ class Game
+    float gameDeltaTime;
+    float gameDeltaTimeMultiplier;
+    
+    
```
Зачем так много отступов? Достаточно одного)

Jass2100 commented on Engine/InGame/GameObject/Actor.h:
```diff
@@ -18,14 +18,18 @@ class Actor : public GameObject
+    float actorDeltaTimeMultiplier = 0;
 
+    virtual void onUpdate();
+    float GetActorDeltaTime();
```
Если это и правда протектед метод, то с маленькой

nikchernyakov replied:
У тебя приватный метод с большой буквы. Но вообще он должен быть публичным

nikchernyakov replied:
В конце нет отступа

nikchernyakov commented on Engine/InGame/Game.h:
```diff
@@ -19,11 +19,18 @@ class Game
     GameAssetManager* gameAssetManager;
 
+    float gameDeltaTime;
+    float gameDeltaTimeMultiplier;
```
deltaTimeMultipler = 1 должен быть

nikchernyakov commented on Engine/InGame/GameObject/Actor.h:
```diff
@@ -18,14 +18,18 @@ class Actor : public GameObject
 
-    virtual void onUpdate();
+    float actorDeltaTime = 0;
+    float actorDeltaTimeMultiplier = 0;
```
deltaTimeMultipler = 1 должен быть

nikchernyakov replied:
Здесь нужно 1 присвоить, ты же не хочешь, чтобы эктор был остановлен в самом начале игры

nikchernyakov commented on Engine/InGame/GameObject/Actor.h:
```diff
@@ -18,14 +18,18 @@ class Actor : public GameObject
     std::vector<ActorComponent*> components;
 
-    virtual void onUpdate();
+    float actorDeltaTime = 0;
```
Здесь 0 не обязательно присваивать

MMTsallagova committed:
Fix

- fix code style
- Change public name function in Actor
- Change public name function in Game

nikchernyakov commented on Engine/InGame/Game.h:
```diff
@@ -19,11 +19,16 @@ class Game
 
     virtual void Update();
     virtual void Render();
+    void SetGameDeltaTimeMultiplier(float deltaTimeMultiplier);
```
Отдели строчкой от функций выше

nikchernyakov commented on Engine/InGame/Game.cpp:
```diff
@@ -32,3 +32,14 @@ void Game::Render()
+
+void Game::SetGameDeltaTimeMultiplier(float deltaTimeMultiplier)
+{
+    gameDeltaTimeMultiplier = deltaTimeMultiplier;
```
Никак не обрабатываешь случай, когда deltaTimeMultiplier <0 или >1 
Делай clamp (есть в нашей библиотеке MathUtils

nikchernyakov commented on Engine/InGame/Game.h:
```diff
@@ -19,11 +19,16 @@ class Game
     GameAssetManager* gameAssetManager;
 
+    float gameDeltaTime;
+    float gameDeltaTimeMultiplier = 1;
```
Добавь ещё getter для этой переменной тоже

nikchernyakov commented on Engine/InGame/GameObject/Actor.cpp:
```diff
@@ -47,3 +53,8 @@ void Actor::AddComponent(ActorComponent* component)
+
+void Actor::SetActorDeltaTimeMultiplier(float deltaTimeMultiplier)
+{
+    actorDeltaTimeMultiplier = deltaTimeMultiplier;
```
Никак не обрабатываешь случай, когда deltaTimeMultiplier <0 или >1 
Делай clamp (есть в нашей библиотеке MathUtils

nikchernyakov commented on Engine/InGame/GameObject/Actor.h:
```diff
@@ -18,14 +18,18 @@ class Actor : public GameObject
     std::vector<ActorComponent*> components;
 
-    virtual void onUpdate();
+    float actorDeltaTime = 1;
```
Здесь не нужно вообще ничего присваивать, как ты в game делаешь

nikchernyakov commented on Engine/InGame/GameObject/Actor.h:
```diff
@@ -18,14 +18,18 @@ class Actor : public GameObject
 
-    virtual void onUpdate();
+    float actorDeltaTime = 1;
+    float actorDeltaTimeMultiplier = 1;
```
Добавь ещё getter для этой переменной тоже

MMTsallagova committed:
Add getters and clamp

- Add getters for Actor and Game
- Add clamp

nikchernyakov commented on Engine/InGame/GameObject/Actor.cpp:
```diff
@@ -47,3 +58,8 @@ void Actor::AddComponent(ActorComponent* component)
+
+void Actor::SetActorDeltaTimeMultiplier(float deltaTimeMultiplier)
+{
+    actorDeltaTimeMultiplier = std::clamp(deltaTimeMultiplier, 0.0f, 1.0f);;
```
Лишняя точка с запятой

nikchernyakov commented on Engine/InGame/GameObject/Actor.h:
```diff
@@ -18,14 +18,19 @@ class Actor : public GameObject
     Transform* GetTransform() const;
 
     void AddComponent(ActorComponent* component);
+    void SetActorDeltaTimeMultiplier(float deltaTimeMultiplier);
```
Отдели от Add Component строкой

nikchernyakov commented on Engine/InGame/GameObject/Actor.h:
```diff
@@ -18,14 +18,19 @@ class Actor : public GameObject
     void AddComponent(ActorComponent* component);
+    void SetActorDeltaTimeMultiplier(float deltaTimeMultiplier);
+    float GetActorDeltaTime();
+    float GetActorDeltaTimeMultiplier();
```
Поменяй местами с функцией выше

nikchernyakov commented on Engine/InGame/GameObject/Actor.cpp:
```diff
@@ -47,3 +58,8 @@ void Actor::AddComponent(ActorComponent* component)
     components.push_back(component);
 }
+
+void Actor::SetActorDeltaTimeMultiplier(float deltaTimeMultiplier)
```
Расположение функций бы тут тоже поправить

nikchernyakov commented on Engine/InGame/Game.h:
```diff
@@ -20,10 +20,17 @@ class Game
 
+    void SetGameDeltaTimeMultiplier(float deltaTimeMultiplier);
+    float GetGameDeltaTime();
+    float GetGameDeltaTimeMultiplier();
```
Тоже поменять местами с геттером выше

MMTsallagova committed:
Swap functions

- swap functions
- delete extra character

MMTsallagova committed:
Add checking deltaTime value

- Add checking deltaTime value
